### PR TITLE
Compositor: range loop refactoring

### PR DESCRIPTION
**Pull request by Joilnen**

### CODE REVIEW

File: OgreMain/src/OgreCompositor.cpp [hunk 1]
```diff
@@ -163,24 +163,21 @@ void Compositor::createGlobalTextures()
...
     typedef std::set<String> StringSet;
     StringSet globalTextureNames;
 
     //Initialize global textures from first supported technique
     CompositionTechnique* firstTechnique = mSupportedTechniques[0];
 
-    const CompositionTechnique::TextureDefinitions& tdefs = firstTechnique->getTextureDefinitions();
-    CompositionTechnique::TextureDefinitions::const_iterator texDefIt = tdefs.begin();
-    for (; texDefIt != tdefs.end(); ++texDefIt)
+    for (const auto& t : firstTechnique->getTextureDefinitions())
```
Comment (paroj) [inline OgreMain/src/OgreCompositor.cpp:172]:
```suggestion
    for (auto def : firstTechnique->getTextureDefinitions())
```

File: OgreMain/src/OgreCompositor.cpp [hunk 2]
```diff
@@ -252,19 +248,16 @@ void Compositor::createGlobalTextures()
     }
 
     //Validate that all other supported techniques expose the same set of global textures.
-    for (size_t i=1; i<mSupportedTechniques.size(); i++)
```
Comment (paroj) [inline OgreMain/src/OgreCompositor.cpp:255]: this is not equivalent

File: OgreMain/src/OgreCompositor.cpp [hunk 3]
```diff
@@ -252,19 +248,17 @@ void Compositor::createGlobalTextures()
...
+    for (size_t i = 1; i < mSupportedTechniques.size(); i++)
     {
-        CompositionTechnique* technique = mSupportedTechniques[i];
         bool isConsistent = true;
         size_t numGlobals = 0;
-        const CompositionTechnique::TextureDefinitions& tdefs2 = technique->getTextureDefinitions();
-        texDefIt = tdefs2.begin();
-        for (; texDefIt != tdefs2.end(); ++texDefIt)
+        CompositionTechnique* technique = mSupportedTechniques[i];
+        for (const auto& t2 : technique->getTextureDefinitions())
```
Comment (paroj) [inline OgreMain/src/OgreCompositor.cpp:256]:
```suggestion
        for (auto* texDef : technique->getTextureDefinitions())
```

Comment (Joilnen) [reply OgreMain/src/OgreCompositor.cpp]: done

**Pull request merged**